src/effects/VST/VSTEffect.cpp

```diff
@@ -911,6 +911,139 @@ bool VSTEffect::InitializePlugin()
    return true;
 }
 
+
+bool VSTEffectWrapper::TransferSettingsContents
+(
+   VSTEffectSettings& src,
+   VSTEffectSettings& dst,
+   bool doMove,
+   bool doMerge
+)
+{
+   dst.mNumParams = src.mNumParams;
+   dst.mUniqueID  = src.mUniqueID;
+   dst.mVersion   = src.mVersion;
+
+   assert(dst.mParamsMap.size() == src.mParamsMap.size());
+
+   // Do the chunk first
+   if (src.mChunk)
+   {
+      dst.mChunk = src.mChunk;
+      if (doMove)
+      {
+         src.mChunk = std::nullopt;
+      }
+   }
+   else if(!doMerge)
+   {
+      dst.mChunk = std::nullopt;
+   }
+   
+
+   // Then do an in-place rewrite of dst, avoiding allocations
+   auto& dstMap = dst.mParamsMap;
+   auto dstIter = dstMap.begin(), dstEnd = dstMap.end();
+   auto& srcMap = src.mParamsMap;
+
+   for (auto& [key, srcValue] : srcMap)
+   {
+      assert(dstIter != dstEnd);
+      auto& [dstKey, dstValue] = *dstIter;
+      assert(dstKey == key);
+
+      if (srcValue)
+      {
+         dstValue = srcValue;
+
+         if (doMove)
+         {
+            srcValue = std::nullopt;
+         }
+      }
+      else if (!doMerge)
+      {
+         // source has no value, and we do not want to merge (i.e. keep any
+         // dest value that might be there) - delete the destination value
+         dstValue = std::nullopt;
+      }
+
+      dstIter++;
+   }   
+
+   return true;
+}
+
+
+bool VSTEffectWrapper::CopySettingsContents(const VSTEffectSettings& src,
+                                                  VSTEffectSettings& dst) const
+{
+   return TransferSettingsContents(
+      const_cast<VSTEffectSettings&>(src), dst, /*move =*/ true, /*merge =*/ false);
```

Comment:
CopySettingsContents should not move from src!  src is not declared rvalue reference, so a function should not surprisingly mutate it.

(But ultimately CopySettingsContents will not be needed)

Reply:
Whoops! definitively a copy-paste error there. Thanks.